### PR TITLE
Caplin: Fixed single validator performance 

### PR DESCRIPTION
**Pull request by Giulio2002**
Issues: 
- Had to fix an hack I added for a race
- Basically we need to process attestations otherwise the network kicks us out. However, we can rate limit the stream.
- Fixed switching ENRs logic

### CODE REVIEW

File: cl/phase1/network/services/attestation_service.go [hunk 1]
```diff
@@ -284,47 +283,47 @@ func (s *attestationService) ProcessMessage(ctx context.Context, subnet *uint64,
 	return ErrIgnore
 }
 
-type attestationJob struct {
-	att          *AttestationWithGossipData
-	creationTime time.Time
-	subnet       uint64
-}
+// type attestationJob struct {
```
Comment (domiwei) [inline cl/phase1/network/services/attestation_service.go:286]: Can we just remove these unused code?

File: cl/phase1/network/services/attestation_service.go [hunk 2]
```diff
@@ -98,7 +98,7 @@ func NewAttestationService(
 		attestationProcessed:     lru.NewWithTTL[[32]byte, struct{}]("attestation_processed", validatorAttestationCacheSize, epochDuration),
 	}
 
-	go a.loop(ctx)
+	//go a.loop(ctx)
```
Comment (domiwei) [inline cl/phase1/network/services/attestation_service.go:101]: also just delete this?

File: cl/beacon/handler/block_production.go [hunk 3]
```diff
@@ -112,6 +112,26 @@ func (a *ApiHandler) waitUntilHeadStateAtEpochIsReadyOrCountAsMissed(ctx context
 		time.Sleep(30 * time.Millisecond)
 	}
 }
+
+func (a *ApiHandler) waitForHeadSlot(slot uint64) {
+	stopCh := time.After(time.Second)
+	for {
```
Comment (domiwei) [inline cl/beacon/handler/block_production.go:118]: I suggest to have a timeout to prevent it from non-stop busy waiting

**Review by domiwei — APPROVED** [review]

File: cl/phase1/network/services/attestation_service.go (hunk 2 above)
Comment (Giulio2002) [reply cl/phase1/network/services/attestation_service.go:101]: will leave it commented for now

File: cl/phase1/network/services/attestation_service.go (hunk 1 above)
Comment (Giulio2002) [reply cl/phase1/network/services/attestation_service.go:286]: will leave it commented for now, in case we will be needing this again

**Pull request merged**